### ISSUE TITLE
deserializing event payload with non backward compatible change uses a default values

### ISSUE THREAD

**Issue by aludwiko** [lightbend/kalix-jvm-sdk#1913]
Here is what I've noticed. I persisted an event payload with schema:
```
message CustomerState {
  string customer_id = 1;
  string email = 2;
  string name = 3;
  Address address = 4;
}
message CustomerCreated {
  CustomerState customer = 1;
}
```
then changed the schema to (string customer_id = 1 -> int32 customer_id):
```
message CustomerState {
  int32 customer_id = 1;
  string email = 2;
  string name = 3;
  Address address = 4;
}
```
and deserialization is just fine, with no exceptions, but `customer_id` is 0, a default value:
```
CustomerState(0,vip@example.com,Very Important,Some(Address(Road 1,The Capital,UnknownFieldSet(Map()))),UnknownFieldSet(Map(1 -> Field(Vector(),Vector(),Vector(),Vector(<ByteString@1486cead size=3 contents="123">)))))
```

I was pretty surprised by this behavior, but apparently, that's how Protobuf works. Users who are not aware might face very bizarre problems because of that. Not sure what we can do about it, except promoting promoting [buf cli](https://buf.build/docs/ecosystem/cli-overview).

**Comment by johanandren** [comment]
Changing the type of an existing field is not a valid change in protobuf, you'd instead have to drop (or mark deprected) the old field and use a field with a new number for the new type. I don't think we do link to it, but a good concise list of guidelines is this: https://protobuf.dev/programming-guides/proto3/#updating

**Comment by johanandren** [comment]
Also surprised that it doesn't cause an error rather btw, since the field number is known but content changed in an incompatible way.

**Comment by aludwiko** [comment]
> Also surprised that it doesn't cause an error rather btw, since the field number is known but content changed in an incompatible way.

Yeah, I also thought that `scalapb.GeneratedMessageCompanion#parseFrom` will throw sth in this case.
